### PR TITLE
Faster ForestFire.jl

### PR DESCRIPTION
**Pull request by Tortar**
This is faster (3x), even if a bit more involved. Inspired by #64

### CODE REVIEW

**Comment by Tortar** [comment]
I kind of feel like ForestFire in the current way is not a good model to test for cellular automata (it should at least start the fire from random points on the grid, not all in a line)

File: ForestFire/Agents/ForestFire.jl [hunk 1]
```diff
@@ -4,24 +4,33 @@ using Agents, Random
...
     return forest, dummystep, tree_step!
 end
 
 function tree_step!(forest)
-    ## Find trees that are burning (coded as 2)
-    @inbounds for I in findall(isequal(2), forest.trees)
-        for idx in nearby_positions(I.I, forest)
-            ## If a neighbor is Green (1), set it on fire (2)
+    k = 0
+    for pos in @view forest.on_fire[1:forest.n[]]
```
Comment (Datseris) [inline ForestFire/Agents/ForestFire.jl:25]: What's going on here? Implementations should only use public API. What is this `n[]`?

**Comment by Tortar** [comment]
it is just a property of the model, even if it has a bad name :D

Il Dom 6 Ago 2023, 21:52 George Datseris ***@***.***> ha
scritto:

**Comment by Datseris** [comment]
See #60 I don't think merging this makes sense. The source code is alien when compared to example codes of ABM implementations in Agents.jl.

**Pull request merged**